### ISSUE TITLE
[Bug Report] Typescript errors in multiple components when enum is used for props

### ISSUE THREAD

**Issue by Giwayume** [element-plus/element-plus#1670]
### Element Plus version
1.0.2-beta.35

### OS/Browsers version
Windows 10 / Chrome 90

### Vue version
3.0.7

### Steps to reproduce
1. Create a Vue typescript project using Vite.
2. Add a component in a Vue template that has a prop which is defined as an enum value, such as <el-tooltip effect="light">
3. Compile the project with npm run build

### What is Expected?
Compilation is successful.

### What is actually happening?
![image](https://user-images.githubusercontent.com/4075314/111858440-81323c80-890f-11eb-802f-eeaebf650a81.png)

Typescript complains about incompatible types, because a string is passed to the prop where it is expecting an enum. A better implementation would be to use union types instead of enums in this scenario. e.g.

Replace the current:
```
export declare enum Effect {
    DARK = "dark",
    LIGHT = "light"
}
```

With:
```
type Effect = "light" | "dark";

<!-- generated by https://elementui.github.io/issue-generator DO NOT REMOVE -->

**Comment by JeremyWuuuuu** [comment]
Yes, because that template cannot connect string literals with enums, we have managed escaping from that via dismissing the compiler type checking, which I think is a problem that we need to fix it.
